src/Behat/Behat/Output/Node/EventListener/JUnit/JUnitFeatureElementListener.php

```diff
@@ -178,16 +135,29 @@ public function printFeatureOnAfterEvent(Formatter $formatter, Event $event)
             );
 
             /** @var AfterStepSetup $afterStepSetup */
-            foreach ($afterScenario['step_setup_events'] as $afterStepSetup) {
+            foreach ($this->afterStepSetupEvents as $afterStepSetup) {
                 $this->setupPrinter->printSetup($formatter, $afterStepSetup->getSetup());
             }
-            foreach ($afterScenario['step_events'] as $afterStepTested) {
+            foreach ($this->afterStepTestedEvents as $afterStepTested) {
                 $this->stepPrinter->printStep($formatter, $afterScenarioTested->getScenario(), $afterStepTested->getStep(), $afterStepTested->getTestResult());
                 $this->setupPrinter->printTeardown($formatter, $afterStepTested->getTeardown());
             }
+
+            $this->afterStepTestedEvents = array();
+            $this->afterStepSetupEvents = array();
         }
+    }
 
-        $this->featurePrinter->printFooter($formatter, $event->getTestResult());
-        $this->afterScenarioTestedEvents = array();
+    /**
+     * Prints the feature on AFTER event.
+     *
+     * @param Formatter $formatter
+     * @param Event     $event
+     */
+    private function printFeatureOnAfterEvent(Formatter $formatter, Event $event)
```

Comment:
This method was public before, but not actually part of the interface.
Should this be kept public?

Reply:
I think that's fine if it's not part of the interface, if anyone complains we can revert